src/test/java/com/google/sps/TripDayTest.java

```diff
@@ -63,12 +66,24 @@ public void testTripDayConstructorNullDestination() {
     locations = new ArrayList<>();
     locations.add(TIMES_SQUARE_ID);
 
-    TripDay tripDay = new TripDay(HOTEL_ID, null, locations);
+    TripDay tripDay = new TripDay(HOTEL_ID, null, locations, DATE);
   }
 
   // Test TripDay constructor with null locations
   @Test(expected = IllegalArgumentException.class)
   public void testTripDayConstructorNullLocations() {
-    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, null);
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, null, DATE);
+  }
+
+  // Test TripDay null date
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripDayConstructorNullDate() {
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, locations, null);
+  }
+
+  // Test TripDay empty date
+  @Test(expected = IllegalArgumentException.class)
+  public void testTripDayConstructorEmptyDate() {
+    TripDay tripDay = new TripDay(HOTEL_ID, HOTEL_ID, locations, "");
```

Comment:
You may want to add tests that check whether the methods that convert the TripDay to an Entity and back work.